src/Downloader.ts

```diff
@@ -586,19 +600,27 @@ class Downloader {
   }
 
   private async getCompressedBody(resp: any): Promise<any> {
-    return this.isMimeTypeImage(resp.headers['content-type']) ? await imagemin.buffer(resp.data, imageminOptions) : resp.data;
+    if (isBitmapImageMimeType(resp.headers['content-type'])) {
+      resp.data = await imagemin.buffer(resp.data, imageminOptions.get(resp.headers['content-type']));
+      if (this.webp) {
+        resp.headers.path_postfix = '.webp';
+        resp.headers['content-type'] = 'image/webp';
+      }
+    }
+    return resp;
   }
 
   private getContentCb = async (url: string, handler: any): Promise<void> => {
     logger.info(`Downloading [${url}]`);
     try {
-      if (this.optimisationCacheUrl && this.isImageUrl(url)) {
+      if (this.optimisationCacheUrl && isImageUrl(url)) {
         this.downloadImage(url, handler);
       } else {
-        const resp = await axios(url, this.arrayBufferRequestOptions);
+        let resp = await axios(url, this.arrayBufferRequestOptions);
+        resp = await this.getCompressedBody(resp);
         handler(null, {
           responseHeaders: resp.headers,
-          content: await this.getCompressedBody(resp),
+          content: resp.data,
```

Comment:
I don't think you need that change. `resp` being and object, it should be passed by reference, the modifications you do in `getCompressedBody` should automatically apply and `responseHeaders: resp.headers` should take them in account.